src/c_wrapper.jl

```diff
@@ -16,28 +23,64 @@ mutable struct ScsMatrix{T} <: AbstractSCSType
     n::T
 end
 
+function ScsMatrix(A::ManagedScsMatrix)
+    return ScsMatrix(
+        pointer(A.values),
+        pointer(A.rowval),
+        pointer(A.colptr),
+        A.m,
+        A.n,
+    )
+end
 mutable struct ScsSettings{T} <: AbstractSCSType
     normalize::T # boolean, heuristic data rescaling
     scale::Cdouble # if normalized, rescales by this factor
+    adaptive_scale::T # boolean, whether to adaptively update `scale`
     rho_x::Cdouble # x equality constraint scaling
     max_iters::T # maximum iterations to take
-    eps::Cdouble # convergence tolerance
-    alpha::Cdouble # relaxation parameter
-    cg_rate::Cdouble # for indirect, tolerance goes down like (1/iter)^cg_rate
+    eps_abs::Cdouble # absolute convergence tolerance
+    eps_rel::Cdouble # relative convergence tolerance
+    eps_infeas::Cdouble # infeasible convergence tolerance
+    alpha::Cdouble # Douglas-Rachford relaxation parameter
+    time_limit_secs::Cdouble # time limit in seconds
     verbose::T # boolean, write out progress
     warm_start::T # boolean, warm start (put initial guess in Sol struct)
-    acceleration_lookback::T # acceleration memory parameter
-    write_data_filename::Cstring
+    acceleration_lookback::T # memory for acceleration
+    acceleration_interval::T # interval to apply acceleration
+    write_data_filename::Cstring # if set dump raw problem data to the file
+    log_csv_filename::Cstring # if set log solve data to the file
     ScsSettings{T}() where {T} = new{T}()
 end
 
+function ScsSettings(linear_solver::Type{<:LinearSolver})
+    stgs = ScsSettings{scsint_t(linear_solver)}()
+    scs_set_default_settings(linear_solver, stgs)
+    return stgs
+end
```

Comment:
I've tried hard not to add these extra methods. I find that they just lead to confusion in the long-term. When wrapping C APIs, the easiest thing is to implement the most minimal interface to C.

So please remove this method and call `scs_set_default_settings` in the `_unsafe_scs_solve` method below.

Reply:
ok, how about making a single inner constructor `ScsSettings(linear_solver)`?

Reply:
Done.